python/dgl/graphbolt/impl/torch_based_feature_store.py

```diff
@@ -260,9 +260,11 @@ class DiskBasedFeature(Feature):
     def __init__(self, path: str, metadata: Dict = None):
         super().__init__()
         mmap_mode = "r+"
-        self._tensor = torch.from_numpy(
-            np.load(path, mmap_mode=mmap_mode)
-        ).contiguous()
+        loaded_array = np.load(path, mmap_mode=mmap_mode)
+        assert not np.isfortran(
+            loaded_array
+        ), "DiskBasedFeature only supports C_CONTIGUOUS array."
+        self._tensor = torch.from_numpy(loaded_array).contiguous()
```

Comment:
As non-contiguous data is always converted to contiguous in existing implementation for `TorchBasedFeature`, let's always do such conversion as well for raw numpy data as well. At the same time, throw a warning for the non-C-style data with `dgl_warning`.

Reply:
Features use DiskBasedFeature are typically very large, and converting them to contiguous in memory would consume a significant amount of memory. Users should handle the conversion themselves.